modules/new_profile/jsx/NewProfileIndex.js

```diff
@@ -0,0 +1,244 @@
+   * the date fields must match.
+   * If match, this function will return true.
+   *
+   * @return {boolean}
+   */
+  validateMatchDate() {
+    const data = this.state.formData;
+    let dateMatch = true;
+    if (data.dateTaken !== data.dateTakenConfirm) {
+      dateMatch = false;
+    }
+    let edcMatch = true;
+    if (this.state.configData['edc'] === 'true' &&
+      data.edcDateTaken !== data.edcDateTakenConfirm
+    ) {
+      edcMatch = false;
+    }
+    return dateMatch && edcMatch;
+  }
+
+  /**
+   * Handles form submission
+   *
+   * @param {event} e - Form submition event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+    const match = this.validateMatchDate();
+    if (!match) {
+      this.setState({
+        errMessage: 'Date of Birth or EDC fields must match',
+        isCreated: false,
+      });
+    } else {
+      let formData = this.state.formData;
+      let formObject = new FormData();
+      for (let key in formData) {
+        if (formData[key] !== '') {
+          formObject.append(key, formData[key]);
+        }
+      }
+      fetch(this.props.submitURL, {
+        method: 'POST',
+        cache: 'no-cache',
+        credentials: 'same-origin',
+        body: formObject,
+        })
+      .then((resp) => resp.json())
+      .then((data) => {
+        this.setState({newData: data});
+        this.setState({isCreated: true});
+       });
+    }
+  }
+
+  /**
+   * Set the form data based on state values of child elements/components
+   *
+   * @param {string} formElement - name of the selected element
+   * @param {string} value - selected value for corresponding form element
+   */
+  setFormData(formElement, value) {
+    let formData = JSON.parse(JSON.stringify(this.state.formData));
+    formData[formElement] = value;
+
+    this.setState({formData: formData});
+  }
+
+  render() {
+    // If error occurs, return a message.
+    if (this.state.error) {
+      return <h3>An error occured while loading the page.</h3>;
+    }
+
+    // Waiting for async data to load
+    if (!this.state.isLoaded) {
+      return <Loader/>;
+    }
+    let profile = null;
+    let edc = null;
+    let edcConfirm = null;
+    let project = null;
+    let pscid = null;
+    if (this.state.configData['useProject'] === 'true') {
+      project =
+        <SelectElement
+          name = "project"
+          label = "Project"
+          options = {this.state.configData.project}
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.project}
+          required = {true}
+        />;
+    }
+    if (this.state.configData['edc'] === 'true') {
+      edc =
+        <DateElement
+          name = "edcDateTaken"
+          label = "Expected Date of Confinement"
+          minYear = "2000"
+          maxYear = "2017"
+          onUserInput = {this.setFormData}
+          value = {this.state.formData.edcDateTaken}
+          required = {true}
+        />;
+      edcConfirm =
```

Comment:
is there a reason why you pulled this date element out into its own variable? i think it can go in the `edc` variable as before

Reply:
It looks like these are meant to be two separate fields, one for the date and one for the date confirmation.

Reply:
sure they're separate fields, but i don't see why they need to be defined as separate render props

Reply:
Not sure if this was Shen's intent, but in my opinion, assigning each component to an individual variable adds some clarity and organization to the components. Although, I probably would have organized this render function differently myself 😛

Reply:
@zaliqarosli  I followed the comment to remove empty <div></div> inside the {edc}, that is why I moved the data confirmation outside the {edc} to avoid the empty <div> tags. Now I make them back. But I have to use empty <div>tags.

Reply:
empty div tags are fine if you need them as a parent jsx element to nest all the others. that was my bad if i told you to remove it! divs are only not need if it nests only 1 component